plugins/module_utils/network.py

```diff
@@ -36,6 +47,8 @@ def __init__(self, ip_address, host_mac, switch_mac="00:00:00:00:00:00"):
         # Receiving socket
         self.rs = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self.rs.bind((Network.BROADCAST_ADDR, Network.UDP_RECEIVE_FROM_PORT))
+        if interface:
+            self.rs.setsockopt(socket.SOL_SOCKET, socket.SO_BINDTODEVICE, interface.encode())
```

Comment:
I think you must do this before the `bind` operation.

Reply:
It seems to work before and after the `bind` operation. I'm fine with both ways :-)

Reply:
I think that doing it after there is a chance to receive data from other sender/machine before you restrict it to a single interface.

Reply:
Agree. I'll change that.